### DISCUSSION TITLE
Questions about the setup process

### DISCUSSION THREAD

**Discussion by IvanKarpov-1** [dyc3/steamguard-cli#334] [Q&A]
Can I use `steamguard-cli` and have Mobile Steam Guard at the same time, or do I need to remove the authenticator from my phone?

This question arose because when I try to set up `steamguard-cli` (`steamguard setup`), after being asked to enter 2fa from my device, I get the error `thread 'main' panicked at 'Failed to log in. Account has not been linked.: UnknownEResult(DuplicateRequest)', src/commands/setup.rs:39:14`. When I wrote `-v debug`, after entering 2fa, I saw that the HTTP request was successful and received the `HTTP header x-eresult: 29`.

Also, I would like to note that I have not used Steam Desktop Authenticator before.

Full log output:
```
DEBUG - Args { global: GlobalArgs { username: None, password: None, all: false, mafiles_path: None, passkey: None, verbosity: Debug, no_update_check: false, http_proxy: None, proxy_credentials: None, danger_accept_invalid_certs: false }, sub: Some(Setup(SetupCommand)), code: CodeCommand { offline: false } }
INFO - reading manifest from /home/mimic/.config/steamguard-cli/maFiles
DEBUG - loading manifest: "/home/mimic/.config/steamguard-cli/maFiles/manifest.json"
DEBUG - passkey was revoked from manifest
DEBUG - Performing auto-upgrade...
DEBUG - Manifest is up to date
Log in to the account that you want to link to steamguard-cli
Username: {MyUsername}
INFO - Logging in to {MyUsername}
DEBUG - prompting for password
Password:
DEBUG - password is present
DEBUG - starting login
DEBUG - HTTP Request: GET https://api.steampowered.com/IAuthenticationService/GetPasswordRSAPublicKey/v1
DEBUG - Response HTTP status: 200 OK
DEBUG - HTTP Header x-eresult: 1
DEBUG - HTTP Request: POST https://api.steampowered.com/IAuthenticationService/BeginAuthSessionViaCredentials/v1
DEBUG - Response HTTP status: 200 OK
DEBUG - HTTP Header x-eresult: 1
DEBUG - auth session started
Enter the 2fa code from your device: FXMY9
DEBUG - HTTP Request: POST https://api.steampowered.com/IAuthenticationService/UpdateAuthSessionWithSteamGuardCode/v1
DEBUG - Response HTTP status: 200 OK
DEBUG - HTTP Header x-eresult: 29
thread 'main' panicked at 'Failed to log in. Account has not been linked.: UnknownEResult(DuplicateRequest)', src/commands/setup.rs:39:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by dyc3** [comment]
While it is technically possible to have the secrets on both steam mobile and steamguard-cli, Steam disabled the ability to back up the app's data a long time ago. So practically, you have to choose one or the other. If you set up steamguard via steamguard-cli, the secrets on your phone will no longer be valid.

This is a new error for me, I haven't seen this before. `DuplicateRequest` makes it seem like it doesn't want you to have 2 active mobile login sessions. Try logging out of the steam mobile app and run setup again.

**Reply by IvanKarpov-1** [reply]
It seems that you still need to [move or delete](https://i.imgur.com/YyF27pk.png) the mobile authenticator to exit the mobile app or set up `steamguard-cli`.

**Reply by dyc3** [reply]
Yes, that's correct. Although it still doesn't explain why you aren't able to complete a login. I'm currently logged in on my steam mobile and in steamguard-cli.

**Reply by IvanKarpov-1** [reply]
I just tried it on my Windows laptop and got the same error.

Can you tell me how you logged into both `steamguard-cli` and the Steam mobile app at the same time? Is it possible that you removed Steam mobile authentication, then set up `steamguard-cli`, and then added mobile authentication again?

I just don't want to remove mobile authentication if possible.

**Reply by dyc3** [reply]
What I did was:

1. Set up the mobile authenticator via steamguard-cli
2. Log in to the account on steam mobile.

I never used the steam mobile app for 2fa.

TBH, I think it's possible that the error you are getting is a bug on steam's side. What country are you in? I've seen the steam API have different behavior depending on the provided language parameter.

**Reply by IvanKarpov-1** [reply]
I'm from Ukraine, and I tried to log in from here on my laptop, although my first attempt was from the US, as I tried to do it from a VPS there.

If you say that you have never used the mobile app for 2FA, then that might explain how you managed to log in both. I think Steam just doesn't allow you to have more than one device to control 2FA.

**Reply by dyc3** [reply]
> Steam just doesn't allow you to have more than one device to control 2FA.

Yes, this is correct. I don't have 2fa codes on steam mobile because it's setup with steamguard-cli.

**Reply by IvanKarpov-1** [reply]
Steam really doesn't allow you to have two devices managing authentication.

I tried to log in a little differently, and this is what the logs say:
```
Enter the 2fa code from your device: jcw53
DEBUG - HTTP Request: POST https://api.steampowered.com/IAuthenticationService/UpdateAuthSessionWithSteamGuardCode/v1
DEBUG - Response HTTP status: 200 OK
DEBUG - HTTP Header x-eresult: 1
INFO - Polling for tokens... -- If this takes a long time, try logging in again.
DEBUG - HTTP Request: POST https://api.steampowered.com/IAuthenticationService/PollAuthSessionStatus/v1
DEBUG - Response HTTP status: 200 OK
DEBUG - HTTP Header x-eresult: 1
INFO - Logged in successfully!
INFO - Adding authenticator...
DEBUG - HTTP Request: POST https://api.steampowered.com/ITwoFactorService/AddAuthenticator/v1
DEBUG - Response HTTP status: 200 OK
DEBUG - HTTP Header x-eresult: 29
ERROR - Failed to link authenticator. Account has not been linked. Authenticator is already present on this account.
ERROR - Authenticator is already present on this account.
DEBUG - Checking for updates to dyc3/steamguard-cli v0.12.2
DEBUG - No update available
```

Now I have another question. If I copy the maFile file to another laptop/PC, will I be able to use the same maFile file to get 2AF codes on both of these devices?

**Reply by dyc3** [reply]
> Now I have another question. If I copy the maFile file to another laptop/PC, will I be able to use the same maFile file to get 2AF codes on both of these devices?

Yes. Steam only allows you to have one set of 2fa secrets. There's nothing stopping you from copying those secrets to other computers.

**Reply by IvanKarpov-1** [reply]
Okay, thank you for your answers. I did remove the authenticator from my phone and now `steamguard-cli` is working. I'm closing the discussion.
